ethToElrond/v2/ethToElrond/steps/step03ProposeTransfer.go

```diff
@@ -0,0 +1,51 @@
+package steps
+
+import (
+	"context"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/core"
+	"github.com/ElrondNetwork/elrond-eth-bridge/ethToElrond/v2/ethToElrond"
+)
+
+type proposeTransferStep struct {
+	bridge ethToElrond.EthToElrondBridge
+}
+
+// Execute will execute this step returning the next step to be executed
+func (step *proposeTransferStep) Execute(ctx context.Context) (core.StepIdentifier, error) {
+	batch := step.bridge.GetStoredBatch()
+
+	wasTransferProposed, err := step.bridge.WasTransferProposedOnElrond(ctx)
+	if err != nil {
+		step.bridge.GetLogger().Error("error determining if the batch was proposed or not on Elrond",
+			"batch ID", batch.ID, "error", err)
+		return ethToElrond.GetPendingBatchFromEthereum, nil
+	}
+
+	if wasTransferProposed {
+		return ethToElrond.SignProposedTransferOnElrond, nil
+	}
+
+	if !step.bridge.MyTurnAsLeader() {
+		return step.Identifier(), nil
```

Comment:
on the architecture graph, if the node is node a leader, it will go to step `get pending`, but actually you set the same step. is it ok?

Reply:
this is ok, as the MyTurnAsLeader method does not error. Updated the graph & TDA document